### ISSUE TITLE
Bug: <short description of the issue>

### ISSUE THREAD

**Issue by divs1210** [typescript-eslint/typescript-eslint#7678]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

parser

### Repro Code

```typescript
const TSParser = require('@typescript-eslint/parser');

const parse = (code) => 
    TSParser.parseForESLint(code).ast.body[0];

parse('(x) => x');

/*
 * ./node_modules/@typescript-eslint/scope-manager/dist/scope/FunctionScope.js:27
 *       const bodyStart = this.block.body?.range[0] ?? -1;
 */
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/<rule-name>": ["error", ...<options>],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

I expected to get an AST, just like:

```javascript
parse('(x) => 1');
```

// gives:

```json
{
  "type": "ExpressionStatement",
  "expression": {
    "type": "ArrowFunctionExpression",
    "generator": false,
    "id": null,
    "params": [
      {
        "type": "Identifier",
        "decorators": [],
        "name": "x",
        "optional": false,
        "loc": {
          "start": {
            "line": 1,
            "column": 1
          },
          "end": {
            "line": 1,
            "column": 2
          }
        }
      }
    ],
    "body": {
      "type": "Literal",
      "value": 1,
      "raw": "1",
      "loc": {
        "start": {
          "line": 1,
          "column": 7
        },
        "end": {
          "line": 1,
          "column": 8
        }
      }
    },
    "async": false,
    "expression": true,
    "loc": {
      "start": {
        "line": 1,
        "column": 0
      },
      "end": {
        "line": 1,
        "column": 8
      }
    }
  },
  "loc": {
    "start": {
      "line": 1,
      "column": 0
    },
    "end": {
      "line": 1,
      "column": 8
    }
  }
}
```

### Actual Result

```
 ./node_modules/@typescript-eslint/scope-manager/dist/scope/FunctionScope.js:27
       const bodyStart = this.block.body?.range[0] ?? -1;
```

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `X.Y.Z` |
| `@typescript-eslint/parser`            | `6.7.2` |
| `@typescript-eslint/rule-tester`       | `X.Y.Z` |
| `@typescript-eslint/scope-manager`     | `X.Y.Z` |
| `@typescript-eslint/typescript-estree` | `X.Y.Z` |
| `@typescript-eslint/type-utils`        | `X.Y.Z` |
| `@typescript-eslint/utils`             | `X.Y.Z` |
| `TypeScript`                           | `X.Y.Z` |
| `ESLint`                               | `X.Y.Z` |
| `node`                                 | `20.6.1` |

**Comment by bradzacher** [comment]
Hi there. A few problems with your issue.

1) you didn't fill out an issue title
2) you didn't actually list anything wrong... there are no bugs or error messages listed.
3) please use the search - duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/7562
